tests/gateway/test_stripe.py

```diff
@@ -0,0 +1,648 @@
+    gateway_params['prefill'] = True
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-email="test@example.com"' in widget.render()
+
+    gateway_params['prefill'] = False
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-email="test@example.com"' not in widget.render()
+
+
+def test_widget_with_remember_me_option(stripe_payment, gateway_params):
+    gateway_params['remember_me'] = True
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-allow-remember-me="true"' in widget.render()
+
+    gateway_params['remember_me'] = False
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-allow-remember-me="false"' in widget.render()
+
+
+def test_widget_with_enable_billing_address_option(stripe_payment, gateway_params):
+    gateway_params['enable_billing_address'] = True
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-billing-address="true"' in widget.render()
+    assert 'data-zip-code="true"' in widget.render()
+
+    gateway_params['enable_billing_address'] = False
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-billing-address="false"' in widget.render()
+    assert 'data-zip-code="false"' in widget.render()
+
+
+def test_widget_with_enable_shipping_address_option(stripe_payment, gateway_params):
+    gateway_params['enable_shipping_address'] = True
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-shipping-address="true"' in widget.render()
+
+    gateway_params['enable_shipping_address'] = False
+    widget = StripeCheckoutWidget(stripe_payment, gateway_params)
+    assert 'data-shipping-address="false"' in widget.render()
+
+
+def test_stripe_payment_form(stripe_payment, gateway_params):
+    form = StripePaymentModalForm(
+        payment=stripe_payment, gateway=None, gateway_params=gateway_params)
+    assert not form.is_valid()
+
+    form = StripePaymentModalForm(
+        data={'stripeToken': FAKE_TOKEN},
+        payment=stripe_payment, gateway=None, gateway_params=gateway_params)
+    assert form.is_valid()
+    with patch.object(stripe_payment, 'charge') as mocked_charge:
+        assert form.process_payment() == stripe_payment
+        mocked_charge.assert_called_once_with(payment_token=FAKE_TOKEN)
+
+
+def test_get_form_class():
+    assert get_form_class() == StripePaymentModalForm
+
+
+def test_get_client(gateway_params):
+    assert _get_client(**gateway_params).api_key == 'secret'
+
+
+def test_get_client_token():
+    assert get_client_token() is None
+
+
+def test_get_error_response_from_exc():
+    stripe_error = \
+        stripe.error.StripeError(json_body=dict(message=ERROR_MESSAGE))
+    invalid_request_error = \
+        stripe.error.InvalidRequestError(message=ERROR_MESSAGE, param=None)
+
+    assert _get_error_response_from_exc(stripe_error) == \
+        dict(message=ERROR_MESSAGE)
```

Comment:
Maybe using dict literals would be more readable? (same below)
```python
    assert _get_error_response_from_exc(stripe_error) == {
        'message': ERROR_MESSAGE}
```

Reply:
Agree, dict literal is also a little bit faster